### ISSUE TITLE
Inform customers that the SSH server on a VM needs to be running on prt 22, in order for the backup service and the monitoring service to work.

### ISSUE THREAD

**Issue by dpkmohan** [rackerlabs/docs-core-infra-user-guide#406]
If customers change the SSH port on their image from port 22 to a different port, the following services may fail:
1. Cloud backup 
2. Cloud monitoring.

If a customer faces these issues on a newly built server that is built from a custom image, it is possible that the custom image has the SSH port changed to a different port. In order to use these services, the custom image must keep its SSH port set to port 22.

**Comment by rose-coste** [comment]
Hi, @dpkmohan, I can think of a few places this may fit:

- troubleshooting advice such as https://developer.rackspace.com/docs/user-guides/infrastructure/cloud-ops/troubleshoot/#troubleshooting-connectivity
- networking concepts such as https://developer.rackspace.com/docs/user-guides/infrastructure/cloud-config/network/cloud-networks-product-concepts/servicenet-publicnet-requirement/#servicenet-publicnet-requirement (also mentions how a network change can cause you to lose access to certain services)
- we could start another troubleshooting topic that points out ways to unintentionally lose access to certain services, now that we know of 2 ways to do it (changing the SSH port, deleting ServiceNet)
- we could add another pre-production step, suggesting to make sure your SSH port is right

Do you have a preference or an alternative suggestion?

**Comment by dpkmohan** [comment]
Hi @rose-coste 
It may make sense to keep this in multiple locations:
1. Troubleshooting:
something like -  "If backup and monitoring agent installations are failing with your custom image..."
2. In the "Creating custom images" section at 
https://developer.rackspace.com/docs/user-guides/infrastructure/cloud-config/compute/cloud-images-product-concepts/custom-images/
something like - "When creating custom images, please note that the SSH service at port 22 must be available in the image, in order to use the monitoring and backup services."

We may not need to add this in Networking, because Cloud Networks doesnt do port blocking yet. This is an OS level setting by the user. 

I need to check to confirm whether points (3) and (4) are true. We know that installation of the service will fail, at server build. I dont know if this affects continued operation..
So 1,2 seem like the baseline set (troubleshooting, and creating custom images). Please let me know what you think.

ps: I'm on PTO the next 2 days. Please anticipate a delay in response.

**Comment by rose-coste** [comment]
Hi, @dpkmohan. I want to point out how someone can check their port assignment for SSH, so in addition to saying that the port assignment should be right we can show how to find out what the assignment is and how to change it if it is wrong. On the way to that, I found other Rackspace doc that could lead someone to create to the problem this addresses. 

In the **Basic cloud server security** article at http://www.rackspace.com/knowledge_center/article/basic-cloud-server-security, in the **SSH config** section, the main text suggests making the SSH port something other than 22 for improved security. A note in tiny letters at the top of the section says not to change the port under some circumstances ("Note: If you have Cloud Servers with a Managed Operations, do not change the default SSH configuration by disabling port 22 because our automation uses this port for access to your server.  If your server does not have Managed Operations, you can choose to implement the following optional security safeguard.") Then the article shows how to change the port. Then it warns that changing the port might not be a great idea ("Note that you should not change the ssh port number if your server has a Managed Operations or if you're using RackConnect."). I can see why a reader could be left not knowing what to do next.

I can work on making that article consistent with whatever we say here, but I think what we say here has to be more complex, pointing out several circumstances under which you *must* keep using port 22, how to determine whether you are in any of those circumstances, and what to expect when keeping port 22 while Rackspace's basic cloud server security advice says its safer not to do that. I think a clean way to connect both documents might be spell out all the complications here, then point the KC article here with a big warning *not* to make the port 22 change without confirming, as explained in the linked-to userguide, that it's the right thing in your situation. 

I'll work out a draft of this so you have something to consider when you're back at work next week.

**Comment by rose-coste** [comment]
I've started drafting this. It's not finished and I'll be out for a few days so it may be the end of the week or early next week before we're ready to publish. You can see what I've done so far at https://github.com/rose-coste/docs-core-infra-user-guide/commit/0d32eb4d8542b79891389dacd8db91800515f18e, a commit to the ssh22 branch of my rose-coste fork. I added network-ssh.rst as a place to discuss this in detail, and added some text and a link to it in security-rst. When published, these are in the "Preparing for productivity" section beginning at https://developer.rackspace.com/docs/user-guides/infrastructure/cloud-preprod/. I also think a link from the troubleshooting section makes sense, so this advice comes up at least twice: when checking everything to make sure it's ready for production work; when looking into why attempts to do production work fail. Please check this so far and let me know if it's going in the right direction.

**Comment by rose-coste** [comment]
@dpkmohan, please see whether what I have staged as a PR at https://github.com/rackerlabs/docs-core-infra-user-guide/pull/409 satisfies this issue:
- details about SSH port 22 are added at cloud-preprod/network-ssh.rst
- troubleshooting guidance at cloud-ops/troubleshoot.rst links to the details about SSH port 22 and is reformatted to create consistency

**Comment by dpkmohan** [comment]
Hi Rose,
Could you tell me how I can see the changes it please? I couldnt get to the .rst files.
Thanks!

Best,
Deepak

**Comment by rose-coste** [comment]
Hi, @dpkmohan, at PR #409 if you click the "Files changed" tab you'll see everything I touched, showing before & after side-by-side; https://github.com/rackerlabs/docs-core-infra-user-guide/pull/409/files takes you right there. 

Another way to do this is to look at the fork (rose-coste) and branch (ssh22) where I made the changes for the PR. That's named in the line of text above the "Files changed" tab: 
rose-coste wants to merge 4 commits into rackerlabs:master from rose-coste:ssh22. https://github.com/rose-coste/docs-core-infra-user-guide/tree/ssh22 takes you right there. 

Another way is that I can send you from here to the new drafts of the two files with most of the changes: 

- https://github.com/rose-coste/docs-core-infra-user-guide/blob/ssh22/cloud-ops/troubleshoot.rst reformats all the troubleshooting guidance and adds a link to the SSH Port 22 details
- https://github.com/rose-coste/docs-core-infra-user-guide/blob/ssh22/cloud-preprod/network-ssh.rst gives the SSH port 22 details. 

Thanks. Let me know if this helps. 
-
